### PR TITLE
SDK image build variant [ESD-243]

### PR DESCRIPTION
**Pull request by silverjam**
Creates a build variant which for an "SDK image" which contains SSH access, but no INS, and no image encryption (and no ability to decrypt protected images).

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.17-32-gf76a5839
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=961490)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-32-gf76a5839&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-32-gf76a5839&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.17-32-gf76a5839
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.17-32-gf76a5839)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.17-32-gf76a5839/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.17-32-gf76a5839

File: fetch_firmware.sh.m4 [hunk 1]
```diff
@@ -63,10 +66,9 @@ download_fw() {
 
   # Download piksi_fpga, try the prod variant first, then sdk variant
   fetch $NAP_S3_PATH_PROD/piksi_prod_fpga.bit $FIRMWARE_DIR/piksi_fpga.bit \
-ifelse(M4_BUCKET, swiftnav-releases,
-`    || fetch $NAP_S3_PATH/piksi_sdk_fpga.bit $FIRMWARE_DIR/piksi_fpga.bit \
-    || error "failed to download piksi_fpga.bit"',
-`    || error "failed to download piksi_fpga.bit"')
+ifelse(M4_BUCKET, swiftnav-releases, `    || fetch_sdk_fpga \
```
Comment (benjaminaltieri) [inline fetch_firmware.sh.m4]: not really sure what is going on here, syntax is foreign to me but I'm guessing there was a reason to refactor the format

File: .travis.yml [hunk 2]
```diff
@@ -32,14 +32,14 @@ matrix:
...
     # Check if docker base image needs to be re-built
-    - env:
-        - TRAVIS_TARGET=docker
-      script:
-        - ./scripts/travis.sh script
-      after_success:
-        - ./scripts/travis.sh after_success
-      after_failure:
-        - ./scripts/travis.sh after_failure
+#    - env:
```
[the remark this answers is not in the document]
Comment (silverjam) [reply .travis.yml]: this is gone in master, so killing here too

File: .travis.yml [hunk 3]
```diff
@@ -72,20 +72,31 @@ matrix:
         - ./scripts/travis.sh after_failure
 
     # Nano image build, for Starling-EVK
+#    - env:
```
[the remark this answers is not in the document]
Comment (silverjam) [reply .travis.yml]: no need to build nano on release branch

File: fetch_firmware.sh.m4 (hunk 1 above)
Comment (silverjam) [reply fetch_firmware.sh.m4]: Removed some duplication and made a helper that generates the `.sha1sum` file that we use to detect if we're doing an SDK build.

**Pull request merged**